src/twisted/conch/test/test_filetransfer.py

```diff
@@ -572,6 +576,46 @@ def test_openDirectoryIteratorDeprecated(self):
         self.assertEqual(message, warnings[0]['message'])
 
 
+    @defer.inlineCallbacks
+    def test_closedConnectionCancelsRequests(self):
+        """
+        If there are requests outstanding when the connection
+        is closed for any reason, they should fail.
+        """
+
+        d = self.client.openFile(b"testfile1", filetransfer.FXF_READ, {})
+        self._emptyBuffers()
+        fh = yield d
+
+        # Intercept the handling of the read request on the server side
+        gotReadRequest = []
+
+        def _slowRead(offset, length):
+            self.assertEqual(gotReadRequest, [])
+            d = defer.Deferred()
+            gotReadRequest.append(d)
+            return d
+        [serverSideFh] = self.server.openFiles.values()
+        serverSideFh.readChunk = _slowRead
+        del serverSideFh
```

Comment:
why `del` here?

Reply:
Ah, just habit to clean up a reference that won't be needed before "blocking" on something later in the method. It makes no difference here.